src/DangerLinqFunctionMapping.ts

```diff
@@ -11,6 +12,9 @@ export const DangerLinqFunctionMapping = {
   [DangerLinqFunction.Aggregate1]: <T>(xs: Array<T>, reducer: (memo: T, val: T) => T): T => {
     throw new DontCallThisError();
   },
+  [DangerLinqFunction.GroupBy]: <T, U>(xs: Array<T>, keySelector: (x: T) => U): Array<Grouping<U, T>> => {
```

Comment:
Kinda looks like I can defined `keySelector` to be a function that returns a non-primitive key. If `U` is an `object`, this might break your implementation according to your gist.

Want to know something fucking crazy? I was looking at how objects can be indexed by objects and functions. Apparently indices get stringified during assignment, so you can actually index an object by one-line functions like this:

![image](https://user-images.githubusercontent.com/31600776/95339360-9ba0be00-0879-11eb-9059-647921121aed.png)


Anyway, I would probably do `<T, U extends string | number>` or smth to keep this simple

Reply:
You know what doesn't have this problem? `Map<T, U>`

Reply:
You might be able to get away with stringifying the result of your `keySelector` fn in your gist?